### PR TITLE
Track `dynamicImportTreeLoadFinished` nodes

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc [hunk 1]
```diff
@@ -824,6 +824,18 @@ ScriptEvaluationResult V8ScriptRunner::EvaluateModule(
...
+    if (recordreplay::DependencyGraphEnabled()) {
+      base::Value::Dict info;
+      info.Set("kind", "evaluateModule");
+      if (module_script)
+        info.Set("url", module_script->SourceUrl().GetString().Utf8());
+      std::string json;
+      base::JSONWriter::Write(info, &json);
+      int node_id = recordreplay::NewDependencyGraphNode(json.c_str());
+      execute.emplace(node_id);
+    }
```
Comment (Andarist) [inline third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc:837]: This is purely a copy-paste (with small adjustments) of the code here: https://github.com/replayio/chromium/blob/656a07519214d4f49536cf0a08a3a66d175d7e6f/third_party/blink/renderer/core/script/pending_script.cc#L192-L207

**Review by Domiii — APPROVED** [review]
The runtime is fickle, not very user-friendly, and, yet, a lot of things require manual testing.
That's why I'd ~~love to see~~ *strongly* suggest to exact notes on how changes were tested, here or in the relevant Linear issue. (Once one gets used to it, its easy to test and easy to reproduce those notes.)

**Comment by Andarist** [comment]
@Domiii I was testing this with: https://github.com/replayio/backend/pull/10501 , just running my tests using the custom build from this PR

**Comment by Domiii** [comment]
> @Domiii I was testing this with: [replayio/backend#10501](https://github.com/replayio/backend/pull/10501) , just running my tests using the custom build from this PR

* @Andarist  That's a link to a PR, no tests and I don't know where to see the test outcomes
* Also not sure if @bhackett1024 can provide guidelines on DG change testing?
  * I do see `RECORD_REPLAY_DEPENDENCY_GRAPH_ASSERTS` (which you can set when recording to get extra asserts going).

**Comment by Andarist** [comment]
> @Andarist That's awesoem! FYI, that's a link to a PR, no tests and I don't know where to see the test outcomes

That's not exactly true. The PR contains tests and I've added a comment there pointing out how it helped: https://github.com/replayio/backend/pull/10501#discussion_r1882352710 . Since then the code has changed though so this was hidden as outdated. At the time of opening this PR here, I had no good idea how to test it - I only learned about it from Brian later.

**Pull request merged**